src/credssp/mod.rs

```diff
@@ -714,6 +692,71 @@ impl SspiImpl for SspiContext {
             SspiContext::CredSsp(credssp) => builder.transform(credssp).execute(),
         }
     }
+
+    fn initialize_security_context_impl<'a>(
+        &'a mut self,
+        builder: &'a mut FilledInitializeSecurityContext<'a, Self::CredentialsHandle>,
+    ) -> GeneratorInitSecurityContext {
+        Generator::new(move |mut yield_point| async move {
+            self.initialize_security_context_impl(&mut yield_point, builder)
+                .await
+        })
+    }
+}
+
+impl<'a> SspiContext {
+    #[instrument(ret, fields(security_package = self.package_name()), skip_all)]
+    async fn change_password_impl(&'a mut self,yield_point: &mut YieldPointLocal, change_password: ChangePassword<'a>) -> crate::Result<()> {
+        match self {
+            SspiContext::Kerberos(kerberos) => kerberos.change_password_impl(yield_point,change_password).await,
+            SspiContext::Negotiate(negotiate) => negotiate.change_password_impl(yield_point,change_password).await,
+            #[cfg(feature = "tsssp")]
+            SspiContext::CredSsp(credssp) => credssp.change_password_impl_async(change_password).await,
+            _ => Err(crate::Error::new(
+                ErrorKind::UnsupportedFunction,
+                "Change password not supported for this protocol",
+            )),
+        }
+    }
+
+    async fn initialize_security_context_impl(
```

Comment:
Put back the `#[instrument]` attribute
```suggestion
    #[instrument(ret, fields(security_package = self.package_name()), skip_all)]
    async fn initialize_security_context_impl(
```